### ISSUE TITLE
Add sideEffects field to package.json

### ISSUE THREAD

**Issue by johncrim** [microsoft/DynamicProto-JS#22]
To make this package optimizeable via [webpack tree shaking](https://webpack.js.org/guides/tree-shaking/), please add a correct `sideEffects` field to the package.json file for this lib.

I care about this b/c the `dynamicproto-js` lib is a transitive dependency of @microsoft/applicationinsights-web . All the other dependencies of @microsoft/applicationinsights-web have this field correctly set.

**Comment by MSNev** [comment]
Which version of App Insights are you looking at as I believe that the AISKU (the source of the applicationinsights-web component) has it as a direct dependency...

And there are no actual side effects created by dynamicProto...
The applicationinsights-shims component (used by AI) does however, so I will add the sideeffects to it which should help the issues with AI complaining about __extends and __assigns.

Or am I missing something?

**Comment by johncrim** [comment]
Thank you for the response, @MSNev  .

I'm using applicationinsights-web@2.5.8, which has these transitive dependencies:

```
├─ @microsoft/applicationinsights-web@2.5.8
│  ├─ @microsoft/applicationinsights-analytics-js@2.5.8
│  ├─ @microsoft/applicationinsights-channel-js@2.5.8
│  ├─ @microsoft/applicationinsights-common@2.5.8
│  ├─ @microsoft/applicationinsights-core-js@2.5.8
│  ├─ @microsoft/applicationinsights-dependencies-js@2.5.8
│  ├─ @microsoft/applicationinsights-properties-js@2.5.8
│  ├─ @microsoft/applicationinsights-shims@1.0.1
│  └─ @microsoft/dynamicproto-js@^1.0.0
```

dynamicproto-js resolves to version 1.0.0.

I looked through all the other transitive dependencies, and they all have `sideEffects` fields in their package.json (applicationinsights-shims has `"sideEffects": false` - sounds like that might be wrong?).

dynamicproto-js is the only transitive dependency that doesn't specify `sideEffects`. It's not in the NPM package, and not in the source here:

https://github.com/microsoft/DynamicProto-JS/blob/master/package.json

Thanks!

**Comment by MSNev** [comment]
Ah, got it. I was looking at the package.json in the lib folder which is not the one that ultimately ends up in the NPM package and as such I misread your initial post (assuming) that the request was to add a specific one for the source file etc.

Will add and publish shortly

**Comment by MSNev** [comment]
Published v1.0.1 with the updated package.json
